contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3608,6 +3608,25 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				GrantStmt *grant = (GrantStmt *) parsetree;
 				char	   *dbname = get_cur_db_name();
 				const char *current_user = GetUserNameFromId(GetUserId(), false);
+				int16 privilege_maskInt = 0;
+				ListCell   *lc1;
+				/* create integer bitmask for permissions*/
+				foreach(lc1, grant->privileges)
+				{
+					AccessPriv *ap = (AccessPriv *) lfirst(lc1);
+					if (!strcmp(ap->priv_name,"execute"))
+						privilege_maskInt |= 32;
+					if (!strcmp(ap->priv_name,"select"))
+						privilege_maskInt |= 16;
+					if (!strcmp(ap->priv_name,"insert"))
+						privilege_maskInt |= 8;
+					if (!strcmp(ap->priv_name,"update"))
+						privilege_maskInt |= 4;
+					if (!strcmp(ap->priv_name,"delete"))
+						privilege_maskInt |= 2;
+					if (!strcmp(ap->priv_name,"references"))
+						privilege_maskInt |= 1;
+				}
```

Comment:
macro. Better to write a function that calculates the permission given privileges list as an input